src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+          }
+        }))
+
+        const transferStateChangeId = await knex('transferStateChange')
+          .select('transferStateChangeId')
+          .where('transferId', transferId)
+          .andWhere('transferStateId', 'COMMITTED')
+          .transacting(trx)
+        if (transferStateChangeId.length === 0 || !transferStateChangeId[0].transferStateChangeId || transferStateChangeId.length > 1) {
+          throw ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to find transfer with COMMITTED state for transferId : ${transferId}`)
+        }
+
+        const participantPositionRecords = await knex('participantPosition')
+          .select('participantPositionId', 'value', 'reservedValue')
+          .where('participantCurrencyId', recordsToInsert[0].participantCurrencyId)
+          .orWhere('participantCurrencyId', recordsToInsert[1].participantCurrencyId)
+          .transacting(trx)
+
+        if (participantPositionRecords.length !== 2) {
+          throw ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to find all participantPosition records for ParticipantCurrency: {${recordsToInsert[0].participantCurrencyId},${recordsToInsert[1].participantCurrencyId}}`)
+        }
+        const participantPositionChangeRecords = participantPositionRecords.map(participantPositionRecord => {
+          participantPositionRecord.transferStateChangeId = transferStateChangeId[0].transferStateChangeId
+          return participantPositionRecord
+        })
+
+        await knex('participantPositionChange')
+          .insert(participantPositionChangeRecords)
+          .transacting(trx)
+
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function insertLedgerEntries (ledgerEntries, transferId, trx = null) {
+  Logger.info(`Ledger entries: ${JSON.stringify(ledgerEntries)}`)
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        for (const ledgerEntry of ledgerEntries) {
+          Logger.info(`Inserting ledger entry: ${JSON.stringify(ledgerEntry)}`)
+          await insertLedgerEntry(ledgerEntry, transferId, trx)
+        }
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function updateTransferSettlement (transferId, status, trx = null) {
```

Comment:
I have some issues with this function:
1. `status` parameter is not used for anything, any idea why it's being passed here?
2. What is the purpose of this function? What is it updating? i.e. I believe this is where all the CGS logic is happening. I feel that this function should be named to reflect this, and additionally, this logic should actually exist in the `domain` layer, with most of the DB calls below be reflective of a model layer function/operation.

Let me know your thoughts on this?

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story